### PR TITLE
Create react-query.md

### PR DESCRIPTION
**Pull request by zfrankdesign**

### CODE REVIEW

File: docs/react/react-query.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# [React Query](https://react-query.tanstack.com/)
+
+## Overview
+
+Kent C. Dodds explains react query's purpose best in his article [Application State Management With React](https://kentcdodds.com/blog/application-state-management-with-react#server-cache-vs-ui-state)
+
+> There are various categories of state, but every type of state can fall into one of two buckets:
+> - Server Cache - State that's actually stored on the server and we store in the client for quick-access (like user data).
+> - UI State - State that's only useful in the UI for controlling the interactive parts of our app (like modal isOpen state).
```
Comment (kevtronkrueger) [inline docs/react/react-query.md:9]: Are you lumping in client state (local storage, cookies, etc) with UI State? Perhaps not worth mentioning...

Comment (zfrankdesign) [reply docs/react/react-query.md:9]: This is just what Kent C. Dodds had in his article. I pretty much just quoted him 😄

File: docs/react/react-query.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+>
+> We make a mistake when we combine the two. Server cache has inherently different problems from UI state and therefore needs to be managed differently. If you embrace the fact that what you have is not actually state at all but is instead a cache of state, then you can start thinking about it correctly and therefore managing it correctly.
```
Comment (kevtronkrueger) [inline docs/react/react-query.md:11]: The meaning of "We make a mistake when we combine the two" is a little ambiguous to me. Would it make more sense to word that like, "when we conflate the source of truth of server data and client state" or "when we attempt to manage server cache within the construct of our UI state or vice versa"?

File: docs/react/react-query.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+>
+> You can definitely manage this yourself with your own useState or useReducer with the right useContext here and there. But allow me to help you cut to the chase and say that caching is a really hard problem (some say that it's one of the hardest in computer science) and it would be wise to stand on the shoulders of giants on this one.
+>
+> This is why I use and recommend [react query](https://react-query.tanstack.com/) for this kind of state. I know I know, I told you that you don't need a state management library, but I don't really consider react query to be a state management library. I consider it to be a cache. And it's a darn good one. Give it a look! That Tanner Linsley is one smart cookie.
```
Comment (kevtronkrueger) [inline docs/react/react-query.md:15]: "I told you that you don't need a state management library". You did no explicitly say this.

You could mention that it's not even feasible to manage anything _but_ server cache with React Query. It forces a clear separation of concern between UI state and server cache.

**Pull request merged**